### PR TITLE
ENG-50: implement confirmation effects (commitReservation, prorateAccrual, updatePaymentSchedule)

### PR DESCRIPTION
**Pull request by Connorbelez**
Implements the three deal-closing confirmation effects that fire when a deal
transitions to confirmed (fundsTransfer.onDone → confirmed):

- commitReservation: commits the pending ledger reservation for the deal
- prorateAccrualBetweenOwners: calculates and writes seller/buyer prorate
  credit entries based on closing date relative to payment schedule
- updatePaymentSchedule: creates a dealReroute record to redirect future
  dispersals from seller to buyer for the transferred share

Adds schema tables (prorateEntries, dealReroutes) and supporting CRUD modules
(mortgages/queries, obligations/queries, prorateEntries/*, dealReroutes/*).
Replaces placeholder registry entries with real effect handlers.

Tests: 9 pass, 6 skipped (convex-test action ctx limitation).

Co-Authored-By: Claude Opus 4.6 (1M context) <noreply@anthropic.com>

## Release Notes

* **New Features**
  * Added payment rerouting capability for ownership transfers during deal closing, enabling funds to flow from seller to buyer for transferred shares.
  * Implemented interest accrual proration calculations to split ownership costs fairly between seller and buyer based on closing dates.
  * Added financial reservation commitment functionality for deal confirmation.

* **Tests**
  * Comprehensive test coverage for deal closing operations, including edge cases and idempotency verification.

* **Documentation**
  * Added implementation gap analysis and acceptance criteria documentation.

### CODE REVIEW

File: convex/engine/effects/dealClosingProrate.ts [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+		const dealId = args.entityId;
+
+		const deal = await ctx.runQuery(internal.deals.queries.getInternalDeal, {
+			dealId,
+		});
+
+		if (!deal) {
+			console.error(`[prorateAccrual] Deal not found: ${dealId}`);
+			return;
+		}
```
[the remark this answers is not in the document]
Comment (Connorbelez) [reply convex/engine/effects/dealClosingProrate.ts]: Same pattern — `getInternalDeal` throws. Null check is defensive, consistent with other effects.
